File: base.lib/src/main/java/com/chiclaim/android/base/BaseActivity.java
package com.chiclaim.android.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;


/**
 * Description：
 * <br/>
 * Created by kumu on 2017/8/3.
 */

public class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return super.onSupportNavigateUp();

        //mActionBar.setNavigationIcon(getResources().getDrawable(R.drawable.ic_action_back));
        //mActionBar.setNavigationOnClickListener
    }


}

File: language-kotlin/kotlin_coroutine/src/main/java/com/chiclaim/coroutine/BaseActivity.kt
package com.chiclaim.coroutine

import androidx.appcompat.app.AppCompatActivity
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlin.coroutines.CoroutineContext

abstract class BaseActivity : AppCompatActivity(), CoroutineScope {

    private val job = Job()

    override val coroutineContext: CoroutineContext
        get() = Dispatchers.Main + job

}